processing/src/main/java/org/apache/carbondata/processing/newflow/DataLoadProcessExecutor.java

```diff
@@ -0,0 +1,28 @@
+package org.apache.carbondata.processing.newflow;
+
+import java.util.Iterator;
+
+import org.apache.carbondata.processing.newflow.steps.DataConverterProcessorStepImpl;
+import org.apache.carbondata.processing.newflow.steps.DataWriterProcessorStepImpl;
+import org.apache.carbondata.processing.newflow.steps.InputProcessorStepImpl;
+import org.apache.carbondata.processing.newflow.steps.SortProcessorStepImpl;
+
+/**
+ * It executes the data load.
+ */
+public final class DataLoadProcessExecutor {
+
+  public void execute(CarbonDataLoadConfiguration configuration, Iterator<Object[]>[] iterators) {
```

Comment:
This function includes creation of the pipeline, so suggest to create a factory for this pipeline. You can call it DataLoadProcessBuilder.build, then execute the pipeline in CarbonLoaderUtil